sktime/forecasting/naive.py

```diff
@@ -139,5 +143,40 @@ def _predict_last_window(self, fh, X=None, return_pred_int=False,
             fh_idx = fh.index_like(self.cutoff)
             return last_window[fh_idx]
 
-        elif self.strategy == "mean":
+        elif self.strategy == "mean" and self.sp == 1:
             return np.repeat(np.nanmean(last_window), len(fh))
+
+        elif self.strategy == "mean" and self.sp != 1:
+            last_window = last_window.astype(float)
```

Comment:
why do we need this?

Reply:
yeah good question! we need this float because using int rounds off the values imputed to padded value(-1) which affects season's mean very much . for example:
mean of 4 and 7 is 5.5 but not using float gives 5.33 here

Reply:
but last window shouldn't be int or is it?

Reply:
last window is int by definition. we are typecasting to float for precision